src/core/include/openvino/opsets/factory.hpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <functional>
+#include <unordered_map>
+
+#include "openvino/core/visibility.hpp"
+
+namespace ov {
+
+/**
+ * @brief Registry of factories that can construct objects derived from BASE_TYPE.
+ */
+template <typename BASE_TYPE>
+class FactoryRegistry {
```

Comment:
as far as I remember, Factory was supposed to be moved to private space, but don't remember details. 
Do you think it's possible ?

Reply:
`ov::OpSet` is only the consumer of `FactoryRegistry`, so I guess we may hide in `opset.cpp` or some other place

Reply:
Remove `FactoryRegistry` class and integrate its most important features into `OpSet` as protected/private part.